clippy_lints/src/blacklisted_name.rs

```diff
@@ -37,6 +38,14 @@ impl_lint_pass!(BlacklistedName => [BLACKLISTED_NAME]);
 
 impl<'tcx> LateLintPass<'tcx> for BlacklistedName {
     fn check_pat(&mut self, cx: &LateContext<'tcx>, pat: &'tcx Pat<'_>) {
+        // Check whether we are under the `test` attribute.
+        if let Some(def_id) = cx.tcx.hir().opt_local_def_id(pat.hir_id) {
```

Comment:
I think this is happening: This is always `None` because a `Pat` is not an item. The test does not lint because `#[test]` is not compiled in ui tests.

Reply:
Should be fixed now. I'm not really happy with the hack required to test the code (not only check for `test`, but also for `ignored`), but I didn't found a more elegant way to do it.

If there is a more correct way, I'd be grateful for any hints.